packages/core/src/expression-builders/where.ts

```diff
@@ -0,0 +1,94 @@
+import type { WhereAttributeHashValue, WhereOperators, AttributeOptions, ColumnReference } from '../model.js';
+import type { Op } from '../operators.js';
+import { BaseSqlExpression } from './base-sql-expression.js';
+import type { Cast } from './cast.js';
+import type { Fn } from './fn.js';
+import type { Literal } from './literal.js';
+
+export type WhereLeftOperand = Fn | ColumnReference | Literal | Cast | AttributeOptions;
+
+/**
+ * Do not use me directly. Use {@link where}
+ */
+export class Where<Operator extends keyof WhereOperators = typeof Op.eq> extends BaseSqlExpression {
+  // TODO [=7]: rename to leftOperand after typescript migration
+  private readonly attribute: WhereLeftOperand;
+  // TODO [=7]: rename to operator after typescript migration
+  private readonly comparator: string | Operator;
+  // TODO [=7]: rename to rightOperand after typescript migration
+  private readonly logic: WhereOperators[Operator] | WhereAttributeHashValue<any> | any;
+
+  constructor(leftOperand: WhereLeftOperand, operator: Operator, rightOperand: WhereOperators[Operator]);
+  constructor(leftOperand: WhereLeftOperand, operator: string, rightOperand: any);
+  constructor(leftOperand: WhereLeftOperand, rightOperand: WhereAttributeHashValue<any>);
+  constructor(
+    leftOperand: WhereLeftOperand,
+    operatorOrRightOperand: string | Operator | WhereAttributeHashValue<any>,
+    rightOperand?: WhereOperators[Operator] | any,
+  ) {
+    super();
+
+    this.attribute = leftOperand;
+
+    if (rightOperand !== undefined) {
+      this.logic = rightOperand;
+      this.comparator = operatorOrRightOperand;
+    } else {
+      this.logic = operatorOrRightOperand;
+      this.comparator = '=';
+    }
+  }
+}
+
+/**
+ * A way of specifying "attr = condition".
+ * Can be used as a replacement for the POJO syntax (e.g. `where: { name: 'Lily' }`) when you need to compare a column that the POJO syntax cannot represent.
+ *
+ * @param leftOperand The left side of the comparison.
+ *  - A value taken from YourModel.rawAttributes, to reference an attribute.
+ *    The attribute must be defined in your model definition.
+ *  - A Literal (using {@link literal})
+ *  - A SQL Function (using {@link fn})
+ *  - A Column name (using {@link col})
+ *  Note that simple strings to reference an attribute are not supported. You can use the POJO syntax instead.
+ * @param operator The comparison operator to use. If unspecified, defaults to {@link OpTypes.eq}.
```

Comment:
We'll have to update this link in the other PR, correct? Or will `OpTypes.eq` remain working?

Reply:
Its implementation drastically changes in the other PR

Reply:
Ah yes, true